src/BugsnagUnity.mm

```diff
@@ -129,11 +129,259 @@
+    }
+    else
+    {
+        ((__bridge BugsnagEvent *)event).severity = BSGSeverityInfo;
+    }     
+}
+
+const char * bugsnag_getSeverityFromEvent(const void *event){   
+    BSGSeverity theSeverity = ((__bridge BugsnagEvent *)event).severity;
+    if(theSeverity == BSGSeverityError)
+    {
+        return strdup("error");
+    }
+    if(theSeverity == BSGSeverityWarning)
+    {
+        return strdup("warning");
+    }
+    return strdup("info"); 
+}
+
+
+void bugsnag_getStackframesFromError(const void *error,const void *instance, void (*callback)(const void *instance, void *stackframes[], int stackframes_size)) {
+
+    NSArray * theStackframes = ((__bridge BugsnagError *)error).stacktrace;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theStackframes.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theStackframes.count; i++) {
+       c_array[i] = (__bridge void *)theStackframes[i];
+    }
+    callback(instance, c_array, [theStackframes count]);
+    free(c_array);
+}
+
+void bugsnag_getErrorsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance, void *errors[], int errors_size)) {
+
+    NSArray * theErrors = ((__bridge BugsnagEvent *)event).errors;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theErrors.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theErrors.count; i++) {
+       c_array[i] = (__bridge void *)theErrors[i];
+    }
+    callback(instance, c_array, [theErrors count]);
+    free(c_array);
+}
+
+void bugsnag_getBreadcrumbsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance, void *breadcrumbs[], int breadcrumbs_size)) {
+
+    NSArray * theBreadcrumbs = ((__bridge BugsnagEvent *)event).breadcrumbs;
+    void **c_array = (void **) malloc(sizeof(void *) * ((size_t)theBreadcrumbs.count));
+    for (NSUInteger i = 0; i < (NSUInteger)theBreadcrumbs.count; i++) {
+       c_array[i] = (__bridge void *)theBreadcrumbs[i];
+    }
+    callback(instance, c_array, [theBreadcrumbs count]);
+    free(c_array);
+}
+
+void bugsnag_getBreadcrumbMetadata(const void *breadcrumb,const void *instance, void (*callback)(const void *instance,const char *keys[], int keys_size, const char *values[], int values_size)) {
+     NSDictionary* sectionDictionary = ((__bridge BugsnagBreadcrumb *)breadcrumb).metadata;
+     NSArray *keys = [sectionDictionary allKeys];
+     NSArray *values = [sectionDictionary allValues];
+     int count = 0;
+     if ([keys count] <= INT_MAX) {
+       count = (int)[keys count];
+     }
+     const char **c_keys = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
+     const char **c_values = (const char **) malloc(sizeof(char *) * ((size_t)count + 1));
+     for (NSUInteger i = 0; i < (NSUInteger)count; i++) {
+       c_keys[i] = [[keys objectAtIndex: i] UTF8String];
+       c_values[i] = [[[values objectAtIndex: i]description] UTF8String];
```

Comment:
Does `description` handle floats and bools reasonably? I'd be concerned that the latter would return 0 or 1.

Maybe it would be possible to serialize the metadata as a JSON string and pass that over to C# for deserialization? That way you can preserve the type information.

Reply:
Created a task to handle this separately 👍